src/client/driver/command-executors/execute-action-command.js

```diff
@@ -0,0 +1,110 @@
+import hammerhead from '../deps/hammerhead';
+import testCafeCore from '../deps/testcafe-core';
+import testCafeRunner from '../deps/testcafe-runner';
+import testCafeUI from '../deps/testcafe-ui';
+import { ElementNotFoundError, InvisibleElementError } from '../../../errors/test-run';
+import COMMAND_TYPE from '../../../test-run/commands/type';
+
+var Promise           = hammerhead.Promise;
+var XhrBarrier        = testCafeCore.XhrBarrier;
+var pageUnloadBarrier = testCafeCore.pageUnloadBarrier;
+var positionUtils     = testCafeCore.positionUtils;
+var domUtils          = testCafeCore.domUtils;
+var waitFor           = testCafeCore.waitFor;
+var ClickAutomation   = testCafeRunner.get('./automation/playback/click');
+var ProgressPanel     = testCafeUI.ProgressPanel;
+
+
+const PROGRESS_PANEL_TEXT   = 'Waiting for the target element of the next action to appear';
+const CHECK_ELEMENT_DELAY   = 200;
+const CHECK_ELEMENT_TIMEOUT = 10000;
+
+
+function ensureElementExists (selector) {
+    return waitFor(selector, CHECK_ELEMENT_DELAY, CHECK_ELEMENT_TIMEOUT)
+        .catch(() => {
+            throw new ElementNotFoundError();
+        });
+}
+
+function ensureElementVisible (element, timeout) {
+    return waitFor(() => positionUtils.isElementVisible(element) ? element : null, CHECK_ELEMENT_DELAY, timeout)
+        .catch(() => {
+            throw new InvisibleElementError(domUtils.getElementDescription(element));
```

Comment:
I wonder if need element description now? We have a callsite now, so there is no big difference between this and  `ElementNotFoundError`